pmd-apex/src/main/resources/category/apex/performance.xml

```diff
@@ -275,4 +275,29 @@ public class Something {
 ]]>
         </example>
     </rule>
+
+    <rule name="OperationWithHighCostInLoop"
+          language="apex"
+          since="7.00.0"
+          message="Avoid operations in loops that may impact performances"
+          class="net.sourceforge.pmd.lang.apex.rule.performance.OperationWithHighCostInLoopRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_performance.html#operationwithhighcostinloop">
+        <description>
+            Schema class methods with high performance cost within loops. Instead, try to call those methods outside the loop.
```

Comment:
I think, this should be a bit more detailed. The name of the rule is very general - which is good on the one side, as the functionality of the rule can be extended in the future. But we should then be very explicit, which kind of problems the rules detect, e.g. currently only `Schema.getGlobalDescribe()` and `Schema.describeSObjects(...)`.

Note, you can use markdown syntax in the description as well.

We might provide links to the official apex reference documentation, like 
* https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_schema.htm#apex_System_Schema_getGlobalDescribe
* https://developer.salesforce.com/docs/atlas.en-us.246.0.apexcode.meta/apexcode/apex_dynamic_global_describe.htm
* https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_schema.htm#apex_System_Schema_describeSObjects

Hopefully, these URLs are valid for a while...

Maybe we can give an explanation, why these methods are expensive performance-wise - although we probably can only guess... reason could be, that getGlobalDescribe() needs to apply permissions, it needs to evaluate _all_ available sObjects in the organization, which can be - depending on the organization - a lot.